### PR TITLE
fix(segment): only emit ionChange when user releases pointer from screen

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: https://github.com/ionic-team/ionic/issues/20257 https://github.com/ionic-team/ionic/issues/20500

## What is the new behavior?

- `ionChange` is now emitted when the user removes their pointer from screen (And if the value has changed)
- This moves segment back in line with the iOS spec, so it is being considered a bug instead of a breaking change.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/segment/segment.tsx [hunk 1]
```diff
@@ -53,15 +57,27 @@ export class Segment implements ComponentInterface {
   @Watch('value')
   protected valueChanged(value: string | undefined) {
     if (this.didInit) {
-      this.ionChange.emit({ value });
+      this.ionSelect.emit({ value });
+      if (!this.pointerDown) {
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply core/src/components/segment/segment.tsx]: Thanks!

File: angular/src/directives/control-value-accessors/value-accessor.ts [hunk 2]
```diff
@@ -12,6 +12,13 @@ export class ValueAccessor implements ControlValueAccessor {
   constructor(protected el: ElementRef) {}
 
   writeValue(value: any) {
+    /**
+     * TODO for Ionic 6:
+     * Change `value == null ? '' : value;`
+     * to `value`. This was a fix for IE9, but IE9
+     * is no longer supported; however, this change
+     * is potentially a breaking change
+     */
```
Comment (juanmiguelbesada) [inline angular/src/directives/control-value-accessors/value-accessor.ts]: It is possible to trigger a deprecation warning if value == null? (console.warn)

File: core/src/components/segment/segment.tsx [hunk 3]
```diff
@@ -51,17 +54,29 @@ export class Segment implements ComponentInterface {
   @Prop({ mutable: true }) value?: string | null;
 
   @Watch('value')
-  protected valueChanged(value: string | undefined) {
-    if (this.didInit) {
-      this.ionChange.emit({ value });
+  protected valueChanged(value: string | undefined, oldValue: string | undefined | null) {
+    this.ionSelect.emit({ value });
+    if (oldValue !== '' || this.didInit) {
```
Comment (juanmiguelbesada) [inline core/src/components/segment/segment.tsx]: Why is needed to check for oldValue?

Comment (liamdebeasi) [reply core/src/components/segment/segment.tsx]: When binding values in Angular using `ngModel`, `ion-segment` value is initialized to `null` and then to the `ngModel` value. We had an old fix in place, but it was causing issues in https://github.com/ionic-team/ionic/issues/20500.

File: angular/src/directives/control-value-accessors/value-accessor.ts (hunk 2 above)
Comment (liamdebeasi) [reply angular/src/directives/control-value-accessors/value-accessor.ts]: There isn't anything to deprecate here as we are just updating the default value for `null`. In Ionic 6 the `null` default would just be changing from `''` to `null`. We will make note of it in the breaking changes document.

**Pull request merged**